app/controllers/friendships_controller.rb

```diff
@@ -0,0 +1,41 @@
+class FriendshipsController < ApplicationController
+  def create
+    @friendship = current_user.friendships.build(friendship_id: params[:format])
+    @friendship.status = 'Pending'
+    return unless @friendship.save
+
+    flash[:notice] = 'Friend Request Sent'
+    redirect_to users_path
+  end
+
+  def update
+    @friendship = Friendship.where("friendship_id = #{current_user.id}"). where("user_id = #{params[:id]}")
+    @friendship.first.status = 'Confirmed'
+    return unless @friendship.first.save
+
+    @friendship = Friendship.create!(user_id: current_user.id, friendship_id: params[:id], status: 'Confirmed')
+    flash[:notice] = 'Friend Request Approved'
+    redirect_to users_path
+  end
+
+  def destroy
+    if params[:method_name] == 'delete'
+      friendship = Friendship.where("friendship_id = #{current_user.id}"). where("user_id = #{params[:id]}")
+      friendship.first.destroy
+      reverse_friendship = Friendship.where("user_id = #{current_user.id}"). where("friendship_id = #{params[:id]}")
+      reverse_friendship.first.destroy
+      flash[:notice] = 'Friend Request Removed'
+    else
+      cancel_friendship = Friendship.where("user_id = #{current_user.id}"). where("friendship_id = #{params[:id]}")
```

Comment:
`params[:id]` was already mentioned by last reviewers as an issue. Read it  here https://github.com/MFahadUmer/ror-social-scaffold/pull/5#discussion_r462677188
You were invited to fix that.